### PR TITLE
[CDAP-2238] Add StreamWriterFactory binding to AbstractTwillRunnable

### PR DESCRIPTION
**Pull request by bdmogal**
``StreamWriterFactory`` binding was missing from ``AbstractTwillRunnable``

Jira: [CDAP-2238](https://issues.cask.co/browse/CDAP-2238)
Build: https://builds.cask.co/browse/CDAP-DUT1593

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramRunnerModule.java [hunk 1]
```diff
@@ -63,6 +63,7 @@ protected void configure() {
     // Create StreamWriter factory.
     install(new FactoryModuleBuilder().implement(StreamWriter.class, DefaultStreamWriter.class)
               .build(StreamWriterFactory.class));
+    expose(StreamWriterFactory.class);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramRunnerModule.java]: Why? Who uses it?

Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramRunnerModule.java]: Not needed. Removed the binding.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java [hunk 2]
```diff
@@ -380,11 +384,25 @@ public Cancellable announce(String serviceName, int port) {
...
     );
   }
 
+  private Module createStreamFactoryModule() {
+    return new PrivateModule() {
+      @Override
+      protected void configure() {
+        install(new FactoryModuleBuilder().implement(StreamWriter.class, DefaultStreamWriter.class)
+                  .build(StreamWriterFactory.class));
+        expose(StreamWriterFactory.class);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java]: If you only have one binding and is always exposed, you don't need to use PrivateModule. Just using AbstractModule is fine.

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java]: But I guess is ok to keep it like this. No need to change.

**Comment by nitinmotgi** [comment]
LGTM :+1:

**Comment by bdmogal** [comment]
Unrelated build failures because of unstable develop. I'm going to merge this, to avoid hogging build resources for this change, since it does not impact any build/unit test scenarios. It only impacts distributed mode, which I have already tested this change on.

**Pull request merged**